pyat/at/plot/standalone.py

```diff
@@ -172,6 +175,112 @@ def plot_sigma(sigma, axis: tuple[str, str] = ('x', 'xp'), scale: float = 1.0,
+                               num_levels=41, plot_separatrix=True, **kwargs):
+    r"""Plot the resulting longitudinal Hamiltonian of a ring (defining the RF bucket).
+    The Hamiltonian is calculated by summing all the cavities in the ring.
+    Harmonic cavities are supported by the function.
+
+    Parameters:
+        ring: Lattice description
+        ct_range (tuple): Forced lower and upper ct values for the plot.
+        Default to :math:`\pm 1.1 \times C / (2h)`
+        dp_range (tuple): Forced lower and upper dp values for the plot.
+        Default to twice the RF acceptance of the bucket.
+        num_points (int): Number of points for 1D grid (ct/dp)
+        Default to 400.
+        num_levels (int): Number of levels for contour plot. Odd number of
+        levels allow to center the colorbar around 0.
+        Default to 41.
+        plot_separatrix (bool): Flag to plot the separatrix contour
+        (:math:`\mathcal{H}=0`).
+
+    Returns:
+        CT:   (num_points,num_points) array: ct grid
+        DP:    (num_points,num_points) array: dp grid
+        hamiltonian:   (num_points,num_points) array: Hamiltonian values
+        along (CT,DP) grid
+    """
+    # Momentum compaction/phase slip factors computed up to third order
+    tmp_ring = ring.disable_6d(copy=True)
+    alpha = get_mcf(tmp_ring, fit_order=3, n_step=10)
+
+    eta = numpy.zeros(len(alpha))
+    eta[0] = alpha[0] - 1 / ring.gamma**2
+    eta[1] = 3 * ring.beta**2 / 2 / ring.gamma**2 + alpha[1] - alpha[0] * eta[0]
+    eta[2] = -ring.beta**2*(5*ring.beta**2 - 1) / (2 * ring.gamma**2) + \
+    alpha[2] -2 * alpha[0] * alpha[1] + alpha[1] / ring.gamma**2 + alpha[0]**2 * eta[0] - \
+    (3 * ring.beta**2 * alpha[0]) / (2 * ring.gamma**2)
+
+    # (ct, dp) grid calculation (defined around the main RF bucket)
+    if ct_range is None:
+        ct = numpy.linspace(-1.1 * ring.circumference/ring.harmonic_number/2,
+                     1.1 * ring.circumference/ring.harmonic_number/2, num=num_points)
+    else:
+        ct = numpy.linspace(ct_range[0], ct_range[1], num=num_points)
+    if dp_range is None:
+        U0 = ring.energy_loss
+        overvoltage = ring.rf_voltage / U0
+        rfa = numpy.sqrt(2 * U0 / (numpy.pi * alpha[0] * ring.harmonic_number * ring.energy) * \
+                      (numpy.sqrt(overvoltage**2 - 1) - numpy.arccos(1 / overvoltage)))
+        dp = numpy.linspace(-2 * rfa, 2 * rfa, num=num_points)
+    else:
+        dp = numpy.linspace(dp_range[0], dp_range[1], num=num_points)
+    CT, DP = numpy.meshgrid(ct, dp)
+
+    # Hamiltonian (H=U+T) divided by harmonic number to have U = U(V_rf, h, phi_s)
+    # First term of the Hamiltonian
+    eta_delta = eta[0] / 2 + eta[1] / 3 * DP + eta[2] / 4 * DP**2
+    T = ring.beta**2 * ring.energy * eta_delta * DP**2
+
+    hamiltonian = T
+    # Iteration over all lattice cavities
+    for rfcav in ring[RFCavity]:
+        Voltage = rfcav.Voltage
+        HarmNumber = rfcav.HarmNumber
```

Comment:
I started using your function with different lattices, and found that this should be:
```python
    Voltage = rfcav.Voltage * ring.periodicity
    HarmNumber = rfcav.HarmNumber * ring.periodicity
```
Alternatively you can replace
`ring.get_revolution_frequency() * HarmNumber` by` rfcav.Frequency`

Reply:
Regarding periodicity, isn't it counterintuitive to set the cavity with an harmonic number divided by the periodicity? Meaning it could lead to non integer harmonic numbers in some cases.
However I think this is a more general comment and not specifically related to this pull request.
I will change the code accordingly.